providers/fcontext.rb

```diff
@@ -0,0 +1,50 @@
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+# Create if doesn't exist, do not touch if fcontext is already registered
+action :add do
+  escaped_file_spec = Regexp.escape(new_resource.file_spec)
+  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
```

Comment:
Ok, I'll do all of that tomorrow.
And splitting at \ wont work. Look at the examples I gave for mysql. Pretty much all regexes I found (semanage fcontext -l) are either .* or (/.*)? at the end. But looking at the output of this command I get the impression that it would be best to explicitly pass the restorecon pathname as an attribute. What do you think?